util/ipgen/renderer.py

```diff
@@ -110,7 +110,8 @@ def _tplfunc_instance_vlnv(self, template_vlnv_str: str):
         if len(template_vlnv) < 3:
             raise TemplateRenderError(
                 f"{template_vlnv_str} isn't a valid FuseSoC VLNV. "
-                "Required format: 'vendor:library:name:version'")
+                "Required format: 'vendor:library:name[:version]'")
+        has_version = len(template_vlnv) == 4
```

Comment:
I wonder whether it would be cleaner to change to `template_vlnv = template_vlnv_str.split(':')` and check that the length isn't greater than 4 either. Otherwise a silly typo will give a version with a colon in it, which might be harder to debug than a "no, don't be stupid!" error.

Reply:
I agree, let me fix that.